src/raft.rs

```diff
@@ -1993,12 +1993,13 @@ impl<T: Storage> Raft<T> {
     fn check_quorum_active(&mut self) -> bool {
         let self_id = self.id;
         let mut act = 0;
+        let learners = self.prs().learner_ids().clone();
         for (&id, pr) in self.mut_prs().iter_mut() {
             if id == self_id {
                 act += 1;
                 continue;
             }
-            if !pr.is_learner && pr.recent_active {
+            if !learners.contains(&id) && pr.recent_active {
```

Comment:
Seems we can split it into two loop for `voters` and `learners` to avoid the query.